### DISCUSSION TITLE
[help] New Channel doesn't see variant and product availability and then new shop can not be available in core v3.8.0-a. (in core v3.5.0-a no problem instead).

### DISCUSSION THREAD

**Discussion by DidoMarchet** [saleor/saleor#10904] [Deployment] [answered]
hi,
we have create two new channel in our Saleor *dashboard v3.1.0-dev core v3.8.0-a*.

We  have configurate the channel as always.

We turn available the products and variants.

And this this is the response:

```
{
  "variant": {
    "id": "UHJvZHVjdFZhcmlhbnQ6NzM=",
    "channelListings": [
      {
        "channel": {
          "id": "Q2hhbm5lbDoy",
          "name": "Channel Emblema",
          "currencyCode": "EUR",
          "__typename": "Channel"
        },
        "price": {
          "amount": 199,
          "currency": "EUR",
          "__typename": "Money"
        },
        "costPrice": null,
        "preorderThreshold": {
          "quantity": null,
          "soldUnits": 0,
          "__typename": "PreorderThreshold"
        },
        "__typename": "ProductVariantChannelListing"
      },
      {
        "channel": {
          "id": "Q2hhbm5lbDox",
          "name": "Channel Aurigane",
          "currencyCode": "EUR",
          "__typename": "Channel"
        },
        "price": {
          "amount": 199,
          "currency": "EUR",
          "__typename": "Money"
        },
        "costPrice": null,
        "preorderThreshold": {
          "quantity": null,
          "soldUnits": 0,
          "__typename": "PreorderThreshold"
        },
        "__typename": "ProductVariantChannelListing"
      }
    ],
    "product": {
      "id": "UHJvZHVjdDoyMw==",
      "channelListings": [
        {
          "isPublished": true,
          "publicationDate": "2022-10-11",
          "isAvailableForPurchase": true,
          "availableForPurchase": "2022-10-11",
          "visibleInListings": true,
          "channel": {
            "id": "Q2hhbm5lbDoy",
            "name": "Channel Emblema",
            "currencyCode": "EUR",
            "__typename": "Channel"
          },
          "__typename": "ProductChannelListing"
        },
        {
          "isPublished": true,
          "publicationDate": "2022-10-11",
          "isAvailableForPurchase": true,
          "availableForPurchase": "2022-10-11",
          "visibleInListings": true,
          "channel": {
            "id": "Q2hhbm5lbDox",
            "name": "Channel Aurigane",
            "currencyCode": "EUR",
            "__typename": "Channel"
          },
          "__typename": "ProductChannelListing"
        }
      ],
      "__typename": "Product"
    },
    "__typename": "ProductVariant"
  },
  "errors": [],
  "__typename": "ProductVariantChannelListingUpdate"
}
```

But product are not visible, in listing.
If I try to query them it return available ONLY FOR ONE CHANNEL (the first created) and not available for the other.

This is the query for the first created Channel 
```
query{
  products(first: 100, channel:"channel-aurigane"){
    edges{
      node{
        name
        isAvailable
        variants{
          quantityAvailable
        }
      }
    }
  }
}
```

```
{
  "data": {
    "products": {
      "edges": [
        {
          "node": {
            "name": "850 Rosso Sole con Aste Nere",
            "isAvailable": true,
            "variants": [
              {
                "quantityAvailable": 50
              }
            ]
          }
        }, ....
```

and this is the query for another channel with the same settings but products non available.
```
query{
  products(first: 100, channel:"channel-emblema"){
    edges{
      node{
        name
        isAvailable
        variants{
          quantityAvailable
        }
      }
    }
  }
}
```

```
{
  "data": {
   "products": {
      "edges": [
        {
          "node": {
            "name": "850 Rosso Sole con Aste Nere",
            "isAvailable": false,
            "variants": [
              {
                "quantityAvailable": 0
              }
            ]
          }
        }, ....
```

In another commerce using saleor  dashboard v3.1.0-dev core v3.5.0-a this problem doesn't occure.

Sorry if I ask hastily but we are stuck with new publication and then I would like to know If there's something I missed.

thanks in advance and kind regards,

Davide

**Comment by Anubiella** [comment]
Good morning everyone!

Yes, as my colleague expressed here above, the issue here is that already from the graphql playground you can see that
Saleor cannot see the stock set previously into the warehouse, as usual.

We have tried to overcome this issue by:
_deactivated/activated the new channel
eliminated/re-created the product
removed/linked again the product to the new channel
removed/re-created the product variants, adding them to the Warehouse and setting their stock and price._

We have, briefly, repeated what we've done before with the previous project, that use 3 channels.
We're having hard times founding some way to cope with this issue,
do you have any guess?

Thanks in advance,
Lorenz

**Comment by fowczarek** [comment]
@DidoMarchet 
Variant and product data look ok. All required channel data are settled. The product is not available for sale due to 0 quantity available.
Did you assign shipping zones and warehouses in the channel configuration?

**Comment by Anubiella** [comment]
Hi @fowczarek ,

yes, every channel is linked with the Shipping Zone (there is just 1 shipping zone in total):

![Channel details - Saleor e-commerce 12-10-2022 12-38-05](https://user-images.githubusercontent.com/31857092/195322567-3ca60fa8-e1a3-4423-abe7-1a34426b9b41.png)


And the shipping zone is linked to the unique warehouse and the three channels:

![Metodi di Spedizione - Saleor e-commerce 12-10-2022 12-37-30](https://user-images.githubusercontent.com/31857092/195322813-1364492d-a7df-447d-9cc8-b6bf88f3ff28.png)

In this case there is only one currency and one country (at the moment).

Are we missing something?
Thanks in advance,
Lorenzo

**Reply by fowczarek** [reply]
Could you also check that warehouse is assigned to the channel? This should be below the place where you could assign the shipping zone to the channel. 
![image](https://user-images.githubusercontent.com/43955230/195812785-0a97c137-8bee-4c8d-b863-6404f6fd0d2b.png)

**Comment by timchesney** [comment]
@Anubiella 
Have you set pricing for each for product for the new channel (at product or variant level depending on the product)? If channel-specific pricing isn't set for each product they generally won't show in any lists. 
Also have you ensured that you have ticked the right boxes under availability? In particular 'Show in product listings'. 

<img width="366" alt="ScreenShot 2022-10-13 at 14 55 21@2x" src="https://user-images.githubusercontent.com/30231/195480732-f72f60e8-e82c-4d25-ba9b-4b37e16e8a85.png">

**Comment by Anubiella** [comment]
Hi @timchesney ,

yes it's all linked and set, like you can see down below (it's in italian, but you can guess more or less):

![850 Rosso Sole con Aste Nere - Saleor e-commerce 13-10-2022 08-53-33](https://user-images.githubusercontent.com/31857092/195528137-d9d2a2e2-da72-494a-97ed-a7016ee87f0c.png)

![Fumo - M - Saleor e-commerce 13-10-2022 08-53-58](https://user-images.githubusercontent.com/31857092/195528187-24adaed1-5602-4043-9243-f98874f6082d.png)

![Fumo - M - Saleor e-commerce 13-10-2022 08-54-16](https://user-images.githubusercontent.com/31857092/195528211-43b3b07c-7762-4eca-b29a-4f706b9e0672.png)

![Fumo - M - Saleor e-commerce 13-10-2022 08-54-32](https://user-images.githubusercontent.com/31857092/195528240-f7e8f57c-e1e8-4258-b2a0-79ca0846eb30.png)

It's very strange

**Reply by timuric** [reply]
I see you have low stock, perhaps there are checkout reservations somewhere, could increase the stock and try to purchase it again

**Reply by timchesney** [reply]
@Anubiella That is very strange. One last 'obvious' thing to check - do you have the channel status set to enabled on the Channel Configuration page? They are disabled by default when you create them. 

<img width="388" alt="ScreenShot 2022-10-14 at 06 32 44@2x" src="https://user-images.githubusercontent.com/30231/195666586-9ee91bb1-ece4-4ee7-a5ec-41a9c86ab5c5.png">

**Reply by DidoMarchet** [reply]
Hi guys @, sorry If I ask again but I would like to know if there's some way to debug or a new release to try to update because we are stuck with the release of new channel. 
We tried to update Saleor to dashboard v3.1.0-dev core v3.8.0-a but nothing change (instead there's some problem with the new graphql playground).

Thanks and kind regards!
Davide

**Comment by Anubiella** [comment]
Hi guys,
I've raised stock quantity in the variant but nothing changed and I also checked the fact that the channel is active like the working one.

![Channel details - Saleor e-commerce 14-10-2022 11-47-10](https://user-images.githubusercontent.com/31857092/195820663-d00fb4c4-d62c-4858-a375-a33e2f327f72.png)


Something else very strange is that querying stocks into the products query in graphql, that field is not even available,
like you can see:

![Playground - https---saleoremblema-quammbase-it-graphql- 14-10-2022 11-58-45](https://user-images.githubusercontent.com/31857092/195820726-3f3aac4c-00e2-42a7-9bd9-56ca2a8226c0.png)


It seems like something is broken down there, I don't know

**Reply by fowczarek** [reply]
@Anubiella which dashboard version do you use? 
You should be able to assign a warehouse to a channel, on channel view. I don't see this on your screenshot. 
![image](https://user-images.githubusercontent.com/43955230/196648092-08157451-e5db-4832-85c4-65534969f7a5.png)

**Reply by DidoMarchet** [reply]
@timchesney dashboard v3.1.0-dev core v3.8.0-a thanks and kind regards!

**Reply by DidoMarchet** [reply]
> @Anubiella which dashboard version do you use? You should be able to assign a warehouse to a channel, on channel view. I don't see this on your screenshot. ![image](https://user-images.githubusercontent.com/43955230/196648092-08157451-e5db-4832-85c4-65534969f7a5.png)

Hi @fowczarek . can you link the version (and the repo) of your version? because we install the last available on Github.
Thanks and kind regards,
Davide

**Reply by fowczarek** [reply]
I use `main` branch from https://github.com/saleor/saleor-dashboard. 
Due to the line from the below link, I think that maybe you use the old `master` branch, which no longer exists in this repo. 
https://github.com/saleor/saleor-dashboard/blob/158188002af9cc0af3cc24697327b30a17549cf7/package.json#L3

**Comment by DidoMarchet** [comment]
Hi, do you have any suggestion or ideas? 
Meanwhile we can try to update Saleor id there's new updates.

Thanks and regards,

Davide

**Comment by DidoMarchet** [comment, marked as answer]
Hi guys thanks for the suggestion of the dashboard misalignment.
There's was an error in our update routine and now all work like a charm.

Kind regards,

Davide

**Comment by DidoMarchet** [comment]
@fowczarek I saw in the new dashborad the variant creator has been removed. Now we have installed dashboard v3.8.0-dev. It was very helpful and I would like to know if it return or not.

Thanks again!
